lib/madness/command_line.rb

```diff
@@ -127,5 +132,49 @@ def build_toc
     def config
       @config ||= Settings.instance
     end
+
+    # Returns the URL where madness is running, suitable for opening in the
+    # browser
+    def server_url
+      scheme = ENV['MADNESS_FORCE_SSL'] ? 'https' : 'http'
+      host = config.bind == '0.0.0.0' ? 'localhost' : config.bind
+      port = config.port
+      
+      "#{scheme}://#{host}:#{port}"
+    end
+
+    # Returns truthy if the server is running. Will attempt to connect 
+    # multiple times. This is designed to assisn in running some code after
+    # the server has launched.
+    def server_running?(retries: 5, delay: 1)
+      connected = false
+      attempts = 0
+
+      begin
+        connected = Socket.tcp(config.bind, config.port)
+      rescue
+        sleep delay
+        retry if (attempts += 1) < retries
+      end
+
```

Comment:
```
ensure
  connected.close
end

```
to close the socket. `connected.close` reads like a pretty darn good expression :) .

Reply:
Added, thanks.